### PR TITLE
Master base properties search 2022 std dam

### PR DESCRIPTION
**Pull request by Polymorphe57**
test

### CODE REVIEW

**Comment by Julien00859** [comment]
is this a draft or is it review ready?

**Comment by Polymorphe57** [comment]
> is this a draft or is it review ready?

Hello Julien, I am just trying to understand the branch and rework a bit the web part. For now, it is still a draft

**Comment by Polymorphe57** [comment]
Hello @std-odoo . Here is a proposal for a ref of your commit that touch some js files in web.

I have tried to remove as many keys as possible and normalize the search model state.
I have forbidden the possibility to use a property fields as a search default.
I have kept the idea of subItems even for property fields.
I think that because of the isChild key the keyboard navigation is no more constistent. We should maybe review that part too and improve that.

Tell me if you think it is fine for you.

**Comment by Polymorphe57** [comment]
I think that the search bar items corresponding to search items of type field_property should be considered as children too and rework a bit the xml (classes)

File: addons/web/static/src/search/search_bar/search_bar.js [hunk 1]
```diff
@@ -70,16 +74,12 @@ export class SearchBar extends Component {
...
-            const field = this.fields[searchItem.fieldName];
-
-            if (field.type === "properties" && searchItem.type === "field") {
-                tasks.push({ id: searchItem.id, prom: this.insertPropertiesSearchItems(searchItem, field, query) });
-            } else if (!subItems[searchItem.id]) {
-                tasks.push({ id: searchItem.id, prom: this.computeSubItems(searchItem, query) });
+        for (const id of expanded) {
+            const searchItem = this.getSearchItem(id);
+            if (searchItem.type === "field" && searchItem.fieldType === "properties") {
+                tasks.push({ id, prom: this.getSearchItemsProperties(searchItem) });
```
Comment (std-odoo) [inline addons/web/static/src/search/search_bar/search_bar.js:80]: is it still search items ? (method is called `getSearchItemsProperties` and not `getSubItemsProperties`)

**Review by std-odoo — COMMENTED** [review]
@Polymorphe57 Hi :) 

Thanks for the ref :) Look a bit more "JS style", I agree about removing key, search default, etc...

About getSearchItemsProperties, isn't it a bit confusing ? (we build search items, but we push them in "task" which contains subItems)

> I have kept the idea of subItems even for property fields.

What do you means ? (properties still generate search items right ? it's only the "search result" that generate sub items ?)

**Pull request closed without merging**